lxd/db/instances.go

```diff
@@ -93,17 +73,18 @@ type Instance struct {
 	LastUseDate  sql.NullTime
 	Description  string `db:"coalesce=''"`
 	Config       map[string]string
-	Devices      map[string]map[string]string
+	Devices      map[string]Device
 	Profiles     []string
 	ExpiryDate   sql.NullTime
 }
 
 // InstanceFilter specifies potential query parameter fields.
 type InstanceFilter struct {
+	ID      *int
 	Project *string
 	Name    *string
-	Node    *string
-	Type    *instancetype.Type `db:"omit=profiles-ref,config-ref,devices-ref"`
+	Node    *string `db:"omit=URIs"` // TODO: Implement multiple joins for URI generation.
```

Comment:
Is this still TODO now we are moving the UsedBy (and hence the need for URL generation) out of the DB generator?

Reply:
This TODO is addressed in my URI branch, so as far as this PR is concerned it's still a TODO.